src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/high-risk-data-breaches/HighRiskBreachLayout.tsx

```diff
@@ -9,55 +9,96 @@ import { ResolutionContainer } from "../ResolutionContainer";
 import { getLocale } from "../../../../../../../functions/universal/getLocale";
+import { FixView } from "../FixView";
+import {
+  HighRiskBreachTypes,
+  getHighRiskBreachesByType,
+} from "./highRiskBreachData";
+import {
+  StepDeterminationData,
+  StepLink,
+  getNextGuidedStep,
+} from "../../../../../../../functions/server/getRelevantGuidedSteps";
+import { getGuidedExperienceBreaches } from "../../../../../../../functions/universal/guidedExperienceBreaches";
 
 export type HighRiskBreachLayoutProps = {
-  pageData: HighRiskBreach;
+  type: HighRiskBreachTypes;
+  subscriberEmails: string[];
+  data: StepDeterminationData;
 };
 
-export function HighRiskBreachLayout({ pageData }: HighRiskBreachLayoutProps) {
+export function HighRiskBreachLayout(props: HighRiskBreachLayoutProps) {
   const l10n = useL10n();
-  const { title, illustration, content, exposedData, type } = pageData;
+
+  const stepMap: Record<HighRiskBreachTypes, StepLink["id"]> = {
+    ssn: "HighRiskSsn",
+    "credit-card": "HighRiskCreditCard",
+    "bank-account": "HighRiskBankAccount",
+    pin: "HighRiskPin",
+    none: "HighRiskPin",
+  };
+
+  const guidedExperienceBreaches = getGuidedExperienceBreaches(
+    props.data.subscriberBreaches,
+    props.subscriberEmails
+  );
+
+  const pageData = getHighRiskBreachesByType({
+    dataType: props.type,
+    breaches: guidedExperienceBreaches,
+  });
+
+  // The non-null assertion here should be safe since we already did this check
+  // in `./[type]/page.tsx`:
+  const { title, illustration, content, exposedData, type } = pageData!;
   const hasBreaches = type !== "none";
 
   return (
-    <ResolutionContainer
-      type="securityRecommendations"
-      title={title}
-      illustration={illustration}
-      cta={
-        <>
-          <Button
-            variant="primary"
-            small
-            // TODO: Add test once MNTOR-1700 logic is added
-            /* c8 ignore next 3 */
-            onPress={() => {
-              // TODO: MNTOR-1700 Add routing logic + fix event here
-            }}
-          >
-            {hasBreaches
-              ? l10n.getString("high-risk-breach-mark-as-fixed")
-              : l10n.getString("high-risk-breach-none-continue")}
-          </Button>
-          {hasBreaches && (
-            <Link
-              // TODO: Add test once MNTOR-1700 logic is added
-              href="/"
-            >
-              {l10n.getString("high-risk-breach-skip")}
-            </Link>
-          )}
-        </>
+    <FixView
+      subscriberEmails={props.subscriberEmails}
+      data={props.data}
+      nextStepHref={
+        getNextGuidedStep(props.data, stepMap[props.type])?.href ?? ""
       }
-      estimatedTime={hasBreaches ? 15 : undefined}
     >
-      <ResolutionContent
-        content={content}
-        exposedData={exposedData}
-        locale={getLocale(l10n)}
-      />
-    </ResolutionContainer>
+      <ResolutionContainer
```

Comment:
All the content in here is the same except for additional indenting, but now wrapped in `<FixView>`, and with the data passed in as props.